### PR TITLE
feat: pass req on 6 arguments

### LINKED ISSUE #55: VKontakteStrategy: verify callback must take 4 or 5 parameters
Привет!

Я использую твой модуль с @nestjs/passport (это обвязка passport для фреймворка nest).

Вот так выглядит мой обработчик:
```ts
import { Strategy, VerifyCallback } from 'passport-vkontakte'
import { PassportStrategy } from '@nestjs/passport'
import { Injectable } from '@nestjs/common'
import { AuthService } from '../auth.service'

@Injectable()
export class VkStrategy extends PassportStrategy(Strategy, 'vk') {
    constructor(private authService: AuthService) {
        super({
            clientID:     1,
            clientSecret: '123',
            callbackURL:  'https://localhost/auth/vk/callback',
            scope: ['status', 'email', 'friends', 'notify']
        })
    }

    async validate (accessToken: string, refreshToken: string, params: string, profile: any, done: VerifyCallback): Promise<any> {
        console.log({ accessToken, refreshToken, params, profile })
        done(null, profile)
    }
}
```

И после прохождения авторизации в ВК получаю: 
```
[ExceptionsHandler] VKontakteStrategy: verify callback must take 4 or 5 parameters +9700ms
Error: VKontakteStrategy: verify callback must take 4 or 5 parameters
    at VkStrategy.passportVerify [as _verify] (/root/backend/node_modules/passport-vkontakte/lib/strategy.js:95:18)
    at /root/backend/node_modules/passport-oauth2/lib/strategy.js:200:24
    at /root/backend/node_modules/passport-vkontakte/lib/strategy.js:177:7
    at passBackControl (/root/backend/node_modules/oauth/lib/oauth2.js:134:9)
    at IncomingMessage.<anonymous> (/root/backend/node_modules/oauth/lib/oauth2.js:157:7)
    at IncomingMessage.emit (events.js:327:22)
    at IncomingMessage.EventEmitter.emit (domain.js:485:12)
    at endReadableNT (_stream_readable.js:1224:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

Я перешел в `node_modules/passport-vkontakte/lib/strategy.js`,
в функции `verifyWrapper` поменял `var arity = verify.length;` на `var arity = 5;`
и все заработало, в своем обработчике я получил нужный ответ.

К сожалению, у меня нет возможности продебажить все самостоятельно, но очень надеюсь, что ты выпустишь фикс :)

P.S.: `console.log(verify.length)` отдает 0

### PR DESCRIPTION
**Pull request by negezor**
Adds `req` without breaking backward compatibility. A lot of cases when a `req` is needed in a handler

### CODE REVIEW

**Comment by stevebest** [comment]
Also, the error message should now read `'VKontakteStrategy: verify callback must take 4, 5 or 6 parameters'`

File: lib/strategy.js [hunk 1]
```diff
@@ -78,14 +78,16 @@ function Strategy(options, verify) {
...
 
     if (params && params.email) {
       profile.emails = [{value: params.email}];
     }
 
     var arity = verify.length;
-    if (arity == 5) {
+    if (arity == 6) {
+      verify(req, accessToken, refreshToken, params, profile, verified);
+    } else if (arity == 5) {
```
Comment (stevebest) [inline lib/strategy.js:90]:
```suggestion
    }
    else if (arity == 5) {
```

**Comment by negezor** [comment]
I added a paragraph in the documentation about the `req`.

**Comment by negezor** [comment]
Do you have any comments?

**Comment by antonlukin** [comment]
I join the problem. It would be great to add someone else to the contributors with the right to release, since such long waits are critical.

**Comment by stevebest** [comment]
Merged and published as `0.5.0`.

Thanks for the patience, everyone.

**Pull request merged**